packages/models-library/src/models_library/user_preferences.py

```diff
@@ -0,0 +1,150 @@
+        cls, preference_name: PreferenceName
+    ) -> type["_BaseUserPreferenceModel"]:
+        preference_class: type[
+            "_BaseUserPreferenceModel"
+        ] | None = cls._registered_user_preference_classes.get(preference_name, None)
+        if preference_class is None:
+            raise NoPreferenceFoundError(preference_name)
+        return preference_class
+
+    @classmethod
+    def get_preference_name(cls) -> PreferenceName:
+        # NOTE: this will be `unique` among all subclasses.
+        # No class inherited from this one, can be defined using the same name,
+        # even if the context is different.
+        return cls.__name__
+
+    @classmethod
+    def get_default_value(cls) -> Any:
+        return (
+            cls.__fields__["value"].default_factory()
+            if cls.__fields__["value"].default_factory
+            else cls.__fields__["value"].default
+        )
+
+    def dict(  # noqa: A003
+        self,
+        *,
+        include: AbstractSet[int | str] | Mapping[int | str, Any] | None = None,
+        exclude: AbstractSet[int | str] | Mapping[int | str, Any] | None = None,
+        by_alias: bool = False,
+        skip_defaults: bool | None = None,
+        exclude_unset: bool = False,
+        exclude_defaults: bool = False,
+        exclude_none: bool = False,
+    ) -> dict[str, Any]:
+        if exclude is None:
+            config_class = getattr(self, "Config", None)
+            exclude_from_serialization: set | None = getattr(
+                config_class, "exclude_from_serialization", None
+            )
+            if exclude_from_serialization:
+                exclude = exclude_from_serialization
+        return super().dict(
+            include=include,
+            exclude=exclude,
+            by_alias=by_alias,
+            skip_defaults=skip_defaults,
+            exclude_unset=exclude_unset,
+            exclude_defaults=exclude_defaults,
+            exclude_none=exclude_none,
+        )
+
+
+class FrontendUserPreference(_BaseUserPreferenceModel):
+    preference_type: PreferenceType = Field(default=PreferenceType.FRONTEND, const=True)
+
+    preference_identifier: PreferenceIdentifier = Field(
+        ..., description="used by the frontend"
+    )
+
+    class Config:
+        exclude_from_serialization: ClassVar[set[str]] = {
+            "preference_identifier",
+            "preference_type",
+        }
+
+
+class UserServiceUserPreference(_BaseUserPreferenceModel):
+    preference_type: PreferenceType = Field(PreferenceType.USER_SERVICE, const=True)
+
+    # NOTE: preferences are stored per service and the version is not considered
+    service_key: ServiceKey = Field(
+        ..., description="the service which manages the preferences"
+    )
+    file_path: Path = Field(..., description="path of the file")
+
+    class Config:
+        exclude_from_serialization: ClassVar[set[str]] = {
+            "preference_type",
+            "service_key",
+        }
+
```

Comment:
so I'll give one more go at this very weird serialization thing.
Are you actually creating this out of the database right? this comes from the utils_preferences.
And then you send this to the frontend via json() method... and that is why you do this very weird construction.

Basically you have a ```UserServiceUserPrerenceInDB``` and another ```UserServiceUserPreferenceGet```
And your problem is that these models are accomodating 2 different usages, one for the DB, one for the API.
Beyond the fact that this is probably violating a one single concern rule, I think it would make more sense if you would create a function that does ```json(exclude={preference_type", "service_key"})``` instead. it would make things clear instead of "magically" changing the behaviour of the json() method.
That is your call. but I find this way of doing very strange.

Reply:
I think I get your message.

What I really want to achieve here is to have a place where I want to define which fields should not included when serialising to the DB (only).
And I would like to keep track of the fields to be excluded, here, where the model is defined.

Wold you add an utility method on each class definition, like `to_db`?